src/python_testing/TC_IDM_1_4.py

```diff
@@ -0,0 +1,167 @@
+
+        self.print_step(1, "Get remote node's MaxPathsPerInvoke")
+        session_parameters = dev_ctrl.GetRemoteSessionParameters(dut_node_id)
+        max_paths_per_invoke = session_parameters.maxPathsPerInvoke
+
+        asserts.assert_greater_equal(max_paths_per_invoke, 1, "Unexpected error returned from unsupported endpoint")
+        asserts.assert_less_equal(max_paths_per_invoke, 65535, "Unexpected error returned from unsupported endpoint")
+
+        self.print_step(2, "Sending `MaxPathsPerInvoke + 1` InvokeRequest if it fits into single MTU")
+        # In practice, it was noticed that we could only fit 57 commands before we hit the MTU limit as a result we
+        # conservatively try putting up to 100 commands into an Invoke request.
+        cap_for_batch_commands = 100
+        number_of_commands_to_send = min(max_paths_per_invoke + 1, cap_for_batch_commands)
+
+        invalid_command_id = 0xffff_ffff
+        list_of_commands_to_send = []
+        for endpoint_index in range(number_of_commands_to_send):
+            invalid_command = Clusters.BasicInformation.Commands.MfgSpecificPing()
+            invalid_command.command_id = invalid_command_id
+
+            list_of_commands_to_send.append(Clusters.Command.InvokeRequestInfo(endpoint_index, invalid_command))
+
+        asserts.assert_greater_equal(len(list_of_commands_to_send), 2,
+                                     "Step 2 is always expected to try sending at least 2 command, something wrong with test logic")
+        try:
+            await dev_ctrl.SendBatchCommands(dut_node_id, list_of_commands_to_send)
+            # If you get the assert below it is likely because cap_for_batch_commands is actually too low.
+            # This might happen after TCP is enabled and DUT supports TCP.
+            asserts.fail("Unexpected success return from sending too many commands")
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, Status.InvalidAction,
+                                 "DUT sent back an unexpected error, we were expecting InvalidAction")
+            self.print_step(2, "DUT successfully failed to process `MaxPathsPerInvoke + 1` InvokeRequests")
+        except ChipStackError as e:
+            chip_error_no_memory = 0x0b
+            asserts.assert_equal(e.err, chip_error_no_memory, "Unexpected error while trying to send InvokeRequest")
+            # TODO it is possible we want to confirm DUT can handle up to MTU max. But that is not in test plan as of right now.
+            # Additionally CommandSender is not currently set up to enable caller to fill up to MTU. This might be coming soon,
+            # just that it is not supported today.
+            self.print_step(2, "DUTs reported MaxPathsPerInvoke + 1 is larger than what fits into MTU. Test step is skipped")
+
+        if max_paths_per_invoke == 1:
+            self.print_step(3, "Skipping test step as max_paths_per_invoke == 1")
```

Comment:
Can you open an issue for either me or you to fix this? The base stuff now has the ability to mark stuff skipped properly in the TH, but I originally built it just for PICS. Need to fix the API to just allow people to mark steps as skipped.

Reply:
And for the skipped steps below.

Reply:
Will do

Reply:
https://github.com/project-chip/connectedhomeip/issues/31139